spec/requests/api/v1/content_negotiation_spec.rb

```diff
@@ -0,0 +1,29 @@
+describe 'Content negotiation', type: :request do
+  it 'passes through requests demanding JSON:API compliant response' do
+    get '/api/v1/locations', headers: { accept: 'application/vnd.api+json' }
+
+    expect(response).to have_http_status(:ok)
+  end
+
+  it 'responds with 406 NotAcceptable to requests demanding non JSON:API compliant reponse' do
+    get '/api/v1/locations', headers: { accept: 'application/json' }
+
+    expect(response).to have_http_status(:not_acceptable)
+  end
+
+  it 'responds with 415 UnsupportedMediaType to requests containing non JSON:API compliant body' do
+    post '/api/v1/locations', params: { user: { name: 'Harry' } },
+                              headers: { accept: 'application/vnd.api+json',
+                                         'content-type': 'application/x-www-form-urlencoded' }
+
+    expect(response).to have_http_status(:unsupported_media_type)
+  end
+
+  it 'skips Content-Type check when request has empty body' do
```

Comment:
Interesting, I guess this helps with debugging from the browser? I always had this requirement enabled for all types of requests to weed out bad configuration on the frontend. Without a body it doesn't really matter I guess?

Reply:
I'm not sure about debugging from the browser, don't you have to specify `Accept` header? Or is there a workaround/Rails-hack with `.json-api` at the end of URL?

For me, the trigger was requirement to specify `Content-Type` header when doing a GET request with curl. Before this change, request was rejected by the server, but since it didn't have the body I felt I shouldn't need to specify `Content-Type`.

Reply:
I reckon it doesn't matter when no request body. Postel's law comes to mind here.